Ds3/Runtime/Network.cs

```diff
@@ -196,14 +196,29 @@ private HttpWebRequest CreateRequest(Ds3Request request, Stream content)
                 httpRequest.ContentLength = request.GetContentLength();
                 if (content != Stream.Null)
                 {
-                    using (var requestStream = httpRequest.GetRequestStream())
+                    var requestStream = httpRequest.GetRequestStream();
+                    if (content.CanSeek && content.Position != 0)
                     {
-                        if (content.CanSeek && content.Position != 0)
+                        content.Seek(0, SeekOrigin.Begin);
+                    }
+                    using (var webResponseStream = new WebResponseStream(content, request.GetContentLength()))
+                    {
+                        try
+                        {
+                            webResponseStream.CopyTo(requestStream, this.CopyBufferSize);
+                        }
+                        catch (Exception ex)
                         {
-                            content.Seek(0, SeekOrigin.Begin);
+                            const string message =
+                                "Bytes to be written to the stream exceed the Content-Length bytes size specified.";
+                            if (ex.Message.Equals(message))
+                            {
+                                throw new Ds3ContentLengthNotMatch(message);
```

Comment:
We will want to include the exception inside of this exception so that way we do not throw it away.